src/AutoMapper/ProfileMap.cs

```diff
@@ -35,7 +35,7 @@ public ProfileMap(IProfileConfiguration profile, IConfiguration configuration)
             ShouldMapField = profile.ShouldMapField ?? configuration?.ShouldMapField ?? (p => p.IsPublic());
             ShouldMapProperty = profile.ShouldMapProperty ?? configuration?.ShouldMapProperty ?? (p => p.IsPublic());
             ShouldUseConstructor = profile.ShouldUseConstructor ?? configuration?.ShouldUseConstructor ?? (c => true);
-            CreateMissingTypeMaps = profile.CreateMissingTypeMaps ?? configuration?.CreateMissingTypeMaps ?? true;
+            CreateMissingTypeMaps = (profile as IConfiguration ?? configuration)?.CreateMissingTypeMaps ?? true;

```

Comment:
No, I don't have a better idea :)

Reply:
Wouldn't `CreateMissingTypeMaps = configuration?.CreateMissingTypeMaps ?? true;` be better since profile is no longer being used for CreateMissingTypeMaps?

Reply:
Does ProfileMap even need `CreateMissingTypeMaps`? If it's removed from `Profile`? This casting is bothering me - there's a deeper design issue here, I think.

Reply:
Yes, those configuration interfaces are problematic. Part of the problem is that you can create a profile that doesn't have a reference to the global configuration. But without refactoring those again, I don't see a better way.

Reply:
Anyway, this PR aside, the version neeeds to be updated for the MyGet build.

Reply:
I see your point. But I don't think it's really propagating. It's more of a quick solution. And I think it's better then what we have now. A setting that's ignored. It's the minimal change required to remove CreateMissingTypeMaps from Profile.